drivers/can/can_esp32_twai.c

```diff
@@ -90,17 +179,44 @@ static int can_esp32_twai_init(const struct device *dev)
 		return err;
 	}
 
+#ifndef CONFIG_SOC_ESP32
+	/*
+	 * TWAI_CLOCK_DIVIDER_REG is incompatible with CAN_SJA1000_CDR for non-ESP32 MCUs
+	 *   - TWAI_CD has length of 8 bits instead of 3 bits
+	 *   - TWAI_CLOCK_OFF at BIT(8) instead of BIT(3)
+	 *   - TWAI_EXT_MODE bit missing (always "extended" = PeliCAN mode)
+	 *
+	 * Resetting the register to 0 to remove the the mode bit set in can_sja_1000_init().
+	 */
+	can_esp32_twai_write_reg32(dev, CAN_SJA1000_CDR, 0U);
+#endif /* !CONFIG_SOC_ESP32 */
+
 	esp_intr_alloc(twai_config->irq_source, 0, can_esp32_twai_isr, (void *)dev, NULL);
 
 	return 0;
 }
 
+#ifndef CONFIG_SOC_ESP32
+#define TWAI_TIMING_MAX_INITIALIZER					\
+	{								\
+		.sjw = 4,						\
+		.prop_seg = 0,						\
+		.phase_seg1 = 16,					\
+		.phase_seg2 = 8,					\
+		.prescaler = (1U << 13)					\
```

Comment:
```suggestion
		.prescaler = BIT(13)						\
```

You could skip the definition and just assign these directly to `timing_max` with a comment - this would reduce the number of `#ifdef`s.

Reply:
Done. And switched to hex numbers, as they also allow to easily spot the register size (in contrast to decimal numbers). I prefer using the `BIT(x)` macro only for actual bit flags, but not to specify the maximum size of a number.